aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+[ LF %s"Issued At: " message-issued-at ]
+[ LF %s"Expiration Time: " message-expiration-time ]
+[ LF %s"Not Before: " message-not-before ]
+[ LF %s"Request ID: " message-request-id ]
+[ LF %s"Resources:" message-resources ]
+
+message-domain = authority
+message-address = "0x" 64*64HEXDIG
+message-statement = 1*( reserved / unreserved / " " )
+message-uri = URI
+message-version = "1"
+message-chain-id = %s"mainnet" / %s"testnet" / %s"devnet" / %s"localnet" / %s"aptos:mainnet" / %s"aptos:testnet" / %s"aptos:devnet" /  %s"aptos:" 1*DIGIT
+message-nonce = 8*( ALPHA / DIGIT )
+message-issued-at = date-time
+message-expiration-time = date-time
+message-not-before = date-time
+message-request-id = *pchar
+message-resources = \\*( LF "- " URI )
+```
+
+### Domain Separating Signing Algorithm
+
+A plaintext construction of a SIWA message must be prefixed with a valid SIWA hash before signing to ensure domain separation across other signatures. The messages must conform to the following expression:
+
+```
+sha3_256(b"SIGN_IN_WITH_APTOS::") || <siwa_message>
+```
+
+### Wallet Standard
+
+The [Wallet Standard](https://github.com/aptos-labs/wallet-standard) is a chain-agnostic set of interfaces and conventions that aim to improve how applications interact with injected wallets.
+
+#### Suggested Features
+
+The features below will be added to the wallet as suggested features. Features will be optional to maintain compatibility with existing wallets.
+
+`aptos:signIn` method to establish a connection and to sign a _Sign in with Aptos_ authentication message.
+
+```ts
+// `input: AptosSignInInput` - inputs with fields to construct an authentication message
+signIn(input: AptosSignInInput): Promise<UserResponse<AptosSignInOutput>>
+```
+
+### Wallet Implementer
+
+To ensure the safety of our users, we want to take a conservative approach when allowing users to sign potentially harmful messages.
+
+#### Bound Fields Verification
+
+The following fields MUST be verified to prevent phishing attacks: `domain` , `address` , `uri` , `version` , and `chainId` . If the fields do not match the signing request, errors must be displayed to the users detailing them out. It is suggested to add additional steps to the approval process, such as an opt-in checkmark field, that confirms the user’s intent if errors are present.
+
+A wallet may also decide to prevent the approval of invalid signing requests.
+
+## Reference Implementation
+
+### Interfaces
+
+#### @aptos-labs/siwa
+
+A new package named `@aptos-labs/siwa` will be created to house the utilities for the _Sign in with Aptos_ standard. These utilities will be responsible for creating messages, verifying messages, and deserializing public key and/or signatures according.
+
+Below are the primary functions that that will be provided as part of the `@aptos-labs/siwa` utilities:
+
+```ts
+/**
+ * Create a SignIn message text from the input following the ABNF format defined in the Sign in with Aptos
```

Comment:
Pro-tip: In technical writing, you always want to refer to the same thing by one name. It can't be either of `SignMessage`, `AptosSignInMessage` or `SignIn` message. You have to pick one and use it consistently so it is drilled down into your readers head and so they know what it means.